algopy_testing/src/algopy_testing/models/itxn.py

```diff
@@ -13,12 +12,12 @@ def __getattr__(self, name: str) -> Callable[[], typing.Any]:
                 "Test context is not initialized! Use `with algopy_testing_context()` to access "
                 "the context manager."
             )
-        if not context.inner_transactions:
+        if not context.inner_transaction_groups:
             raise ValueError(
                 "No inner transaction found in the context! Use `with algopy_testing_context()` "
                 "to access the context manager."
             )
-        itxn = context.inner_transactions[-1]
+        itxn = context.inner_transaction_groups[-1]
```

Comment:
This should get the last transaction of the last group e.g. `context.inner_tranasctions_groups[-1][-1]` but with appropriate checks

Reply:
Additionally we should also implement GITxn in a similar fashion, in GITxn the first argument for each method is the index of the transaction within the last group